modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2829,6 +2852,13 @@ def broadcast_apply(
             Modin DataFrame to broadcast.
         join_type : str, default: "left"
             Type of join to apply.
+        copartition : bool, default: True
```

Comment:
The introduction of this parameter looks like a fix for situations where the function was called on unaligned partitions. Let's separate this into a separate pull request and add tests.

Reply:
It's not the fix you're talking about.

If you'll look at the previous implementation of this method, you'll see that the default behavior was to perform co-partitioning unconditionally. This parameter just gives an ability to provide a hint on whether we can safely skip the co-partitioning if we're sure that the frames are identically partitioned.